Makefile

```diff
@@ -0,0 +1,87 @@
+.PHONY: test
+test: api-test ui-test ## run all Unit Tests
+
+.PHONY: lint
+lint: api-lint ui-lint yaml-lint ## check all Lints
+
+.PHONY: api-check
+api-check: goa-gen api-test generated api-lint ## all API checks
+
+.PHONY: ui-check
+ui-check: ui-test ui-lint ## all UI checks
+
+.PHONY: goa-gen
+goa-gen: ## generate API Design
+	@echo "----------------------------"
+	@echo "-- Generating API Design... "
+	@echo "----------------------------"
+	cd api && goa gen github.com/tektoncd/hub/api/design
+
+	@echo "----------------------------"
+	@echo "- Generating v1 API Design... "
+	@echo "----------------------------"
+	cd api && cd v1 && goa gen github.com/tektoncd/hub/api/v1/design
```

Comment:
how does this work?
`cd api && goa gen github.com/tektoncd/hub/api/design`
when we do this.. does the directory context changes only for that command?
and 
again below it does
`cd api/v1` so each command runs in  a separate context/terminal?

Reply:
Yes the directory content changes only for that command and also the command runs in a separate context